scripts/tasks/check-imports-are-declared-dependencies.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+// Checks that, for all published packages, any `import`ed packages are declared as explicit dependencies.
+// This avoids issues with transitive dependencies being treated as implicit dependencies, which
+// may break tools like Yarn PnP (see: https://github.com/salesforce/lwc/issues/3540)
+
+const path = require('node:path');
+const fs = require('node:fs');
+const { globSync } = require('glob');
+const { init, parse } = require('es-module-lexer');
+const builtInModules = require('builtin-modules');
```

Comment:
Node has this already builtin

https://nodejs.org/api/module.html#modulebuiltinmodules

Reply:
TIL. Fixed. 3b1ee06e1